src/components/ArrowKeyFocusManager.js

```diff
@@ -28,91 +28,95 @@ const defaultProps = {
     shouldExcludeTextAreaNodes: true,
 };
 
-class ArrowKeyFocusManager extends Component {
-    componentDidMount() {
+function ArrowKeyFocusManager(props) {
+    const onArrowUpKeyRef = useRef();
+    const onArrowDownKeyRef = useRef();
+    const mounted = useRef(false);
+    const prevMaxIndex = useRef();
+
+    useEffect(() => {
+        onArrowUpKeyRef.current = () => {
+            if (props.maxIndex < 0) {
+                return;
+            }
+
+            const currentFocusedIndex = props.focusedIndex > 0 ? props.focusedIndex - 1 : props.maxIndex;
+            let newFocusedIndex = currentFocusedIndex;
+
+            while (props.disabledIndexes.includes(newFocusedIndex)) {
+                newFocusedIndex = newFocusedIndex > 0 ? newFocusedIndex - 1 : props.maxIndex;
+                if (newFocusedIndex === currentFocusedIndex) {
+                    // all indexes are disabled
+                    return; // no-op
+                }
+            }
+            props.onFocusedIndexChanged(newFocusedIndex);
+        };
+
+        onArrowDownKeyRef.current = () => {
+            if (props.maxIndex < 0) {
+                return;
+            }
+
+            const currentFocusedIndex = props.focusedIndex < props.maxIndex ? props.focusedIndex + 1 : 0;
+            let newFocusedIndex = currentFocusedIndex;
+
+            while (props.disabledIndexes.includes(newFocusedIndex)) {
+                newFocusedIndex = newFocusedIndex < props.maxIndex ? newFocusedIndex + 1 : 0;
+                if (newFocusedIndex === currentFocusedIndex) {
+                    // all indexes are disabled
+                    return; // no-op
+                }
+            }
+            props.onFocusedIndexChanged(newFocusedIndex);
+        };
+    });
```

Comment:
I don't understand the purpose of this code in a `useEffect` that will be run in every cycle after rendering. Why did we put it in a use effect and use references (`useRef`)?